### PR TITLE
Add sealed keyword to nodes in graphs"

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

File: scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import dotty.dokka.model._
+import dotty.dokka.model.api._
+
+
+class SealedMarksGraphTransformer(using context: DocContext) extends ModuleTransformer:
+  override def apply(original: DModule): DModule =
+    val sealedRelations = getSealedRelations(original.getPackages.get(0))
+    original.updateMembers { m =>
+      m.mapGraphEdges {
+        case Edge(from @ LinkToType(_, fromDri, _), to @ LinkToType(_, toDri, _), _) =>
```
Comment (romanowski) [inline scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala:18]: can we avoid reallocation of all `Edge objects` e.g.:
```
   m.mapGraphEdges {
        case edge @ Edge(from, to, _) =>
          val desc = if sealedRelations.contains(to.dri) then Edge(from, to, "🔒") else edge
      }
```

File: scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          val desc = if sealedRelations.contains(toDri) then "🔒" else ""
+          Edge(from, to, desc)
+      }
+    }
+
+  private def getSealedRelations(c: Member): Set[DRI] =
+    val selfMapping = if c.modifiers.contains(Modifier.Sealed) then Set(c.dri) else Set.empty
+    c.allMembers.flatMap(getSealedRelations).toSet ++ selfMapping
```
Comment (romanowski) [inline scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala:26]: `selfMapping ++ c.allMembers.flatMap(getSealedRelations)` - that way `toSet can be ommited`

File: scala3doc/src/dotty/renderers/DotDiagramBuilder.scala [hunk 3]
```diff
@@ -22,8 +22,8 @@ object DotDiagramBuilder:
       s"""node${id} [id=node${id}, label="${getHtmlLabel(vertex, renderer)}", style="${getStyle(vertex)}"];\n"""
     }.mkString
 
-    val edges = diagram.edges.map { (from, to) =>
-      s"""node${vWithId(from)} -> node${vWithId(to)};\n"""
+    val edges = diagram.edges.map { case Edge(from, to, description) =>
+      s"""node${vWithId(from)} -> node${vWithId(to)} [label="${description}"];\n"""
```
Comment (romanowski) [inline scala3doc/src/dotty/renderers/DotDiagramBuilder.scala:26]: can we only add `labels` if are non-empty? I do not want to increase HTML size more then we  really need

**Review by romanowski — CHANGES_REQUESTED** [review]

File: scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala (hunk 1 above)
Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala:18]: Yeah, my bad

File: scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala (hunk 2 above)
Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/transformers/SealedMarksGraphTransformer.scala:26]: Good point, will change

**Pull request merged**